### PR TITLE
Refactored terminal.py

### PR DESCRIPTION
**Pull request by colin99d**
# Description

We seem to like adding custom flags to terminal.py. This is getting increasingly complex. I simplified this by converting the main function into using argparse. Also, we are integrating the all_scripts command directly into the terminal. This will make our integration tests much easier from now on. Please note that just running -s results in all scripts being ran, while -s stocks results in all scripts with "stocks" in the file path being ran.

I have also integrated a pytest like display of the amount of successes and failures and scripts that failed are printed at the end. Running the scripts also automatically cleans them so that they can be used as integrated tests. Two items are changed :

- "Exit" is added to the end of every script to prevent the workflow from getting stuck 
- "Reset" is removed from the files, unfortunately testing reset adds a lot of complexity, and for now it is easier to run the tests without them.

**Note:** The files themselves are not changed by this process, only the lists generated based on the files.

Things I still need to do before this is ready to push: 

- [x] Don't break on scripts failure
- [x] Introduce a way to keep track of which commands had an error

# How has this been tested?

I ran this through about every edge case I thought of. But @jmaslek is willing to write a thesis for anyone that can find an error.

# Checklist:

- [x] Update [our Hugo documentation](https://gamestonkterminal.github.io/GamestonkTerminal/) following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: terminal.py [hunk 1]
```diff
@@ -305,28 +306,80 @@ def terminal(jobs_cmds: List[str] = None):
...
+            folder = os.path.join(
+                os.path.abspath(os.path.dirname(__file__)), "scripts/"
+            )
+            files = [
+                name
+                for name in os.listdir(folder)
+                if os.path.isfile(os.path.join(folder, name))
+            ]
+            for file in files:
+                if file[-4:] == ".gst":
```
Comment (jmaslek) [inline terminal.py:373]: if file.endswith(".gst")

**Review by jmaslek — COMMENTED** [review]
Looks good to me.  I'll leave to the real devs to confirm :P

**Comment by DidierRLopes** [comment]
If it has @jmaslek stamp on it, I'm fine with this!

**Comment by colin99d** [comment]
@Chavithra and I are having an issue with capture output. Once we can resolve this we should be good to go.

**Comment by Chavithra** [comment]
Do you it it would be interesting to automatically append `exit` to the queue after loading the script ?

**Comment by colin99d** [comment]
> @colin99d Do you it it would be interesting to automatically append `exit` to the queue after loading the script ?

I think that would be very wise.

**Comment by DidierRLopes** [comment]
Is this ready for review?

File: gamestonk_terminal/portfolio/portfolio_view.py [hunk 2]
```diff
@@ -134,7 +134,7 @@ def display_allocation(portfolio: portfolio_model.Portfolio, export: str = ""):
         if gtff.USE_ION:
             plt.ion()
     except Exception as e:
-        print(e)
+        console.print(e)
```
Comment (DidierRLopes) [inline gamestonk_terminal/portfolio/portfolio_view.py:137]: If there's an error we don't want to show or export something.

In any case, I don't think we need this try-except block here

**Review by DidierRLopes — CHANGES_REQUESTED** [review]
Getting an error with `test_stocks_disc.gst`
Screenshot 2022-02-08 at 22 15 46

If we have 2 errors on the same file, I think only one appears.
Screenshot 2022-02-08 at 22 18 25

If possible would be great to have a:
14/14
11/12
9/9

i.e. success/failures per commands within a file.

In addition, I think that "test" could be a flag and we could filter as you have it.. 
And someone could do:
* `python terminal.py scripts/test.gst --test`
* `python terminal.py scripts/ --test`
* `python terminal.py scripts/ --test -f disc`

**Comment by colin99d** [comment]
@DidierRLopes so you would like to see a test flag and a filter flag separate, instead of the combined method I have used ie. `python terminal.py -t stocks`?

**Comment by DidierRLopes** [comment]
> @DidierRLopes so you would like to see a test flag and a filter flag separate, instead of the combined method I have used ie. `python terminal.py -t stocks`?

Kind of. But even a path to the folder being given instead of having a pre-defined

**Comment by DidierRLopes** [comment]
Path should allowed to be a file, and since `-t` flag is enabled it should output success/fail
Screenshot 2022-02-10 at 00 17 56

Apart from this things are looking great 😄

**Review by DidierRLopes — APPROVED** [review]
Let's goooo!

**Pull request merged**